src/core/layout/qgslayoutitemattributetable.cpp

```diff
@@ -480,10 +479,9 @@ bool QgsLayoutItemAttributeTable::getTableContents( QgsLayoutTableContents &cont
     req.setFilterFid( atlasFeature.id() );
   }
 
-  QVector< QPair<int, bool> > sortColumns = sortAttributes();
-  for ( int i = sortColumns.size() - 1; i >= 0; --i )
+  for ( const QgsLayoutTableColumn &column : qgis::as_const( mSortColumns ) )
```

Comment:
I think there's an api break here -- if someone is using the old api to set a list of columns with their sort order contained, then that won't be respected here.

Reply:
Then setColumns would be used, right?
When using `setColumns`, I can check if a sorting order is provided and call `setSortColumns` at the same time.
Works for you?

Reply:
I thought about that too- but wouldn't it be hard to differentiate between a call to setColumns using the old API, vs correct use of the new api and calling setSortColumns and then later setColumns?

Reply:
Well, if you correctly use the API there is no sorting info (rank) in the list when calling setColumns. If there is, then you're using the old API and we call setSortColumns.